common/workunit/workunit.hpp

```diff
@@ -1785,4 +1782,17 @@ extern WORKUNIT_API TraceFlags loadTraceFlags(IConstWorkUnit * wu, const std::in
 
 extern WORKUNIT_API bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, unsigned wfid, const char *graphName);
 
+
+class WORKUNIT_API StatisticsAggregator : public CInterface
+{
+public:
+    StatisticsAggregator() : aggregateKindsMapping(aggregateKindStatistics) {}
+    void loadExistingAggregates(IConstWorkUnit &workunit);
+    void recordStats(IStatisticCollection * sourceStats, unsigned wfid, const char * graphName, unsigned graphId);
+    void updateAggregates(IWorkUnit *wu);
+private:
+    Owned<IStatisticCollection> statsCollection;
+    const StatisticsMapping & aggregateKindsMapping;
```

Comment:
personal/trivial: as this is explicitly an aggregator class, I this would be better without 'aggregate' in its naming (personally I'd call it 'mapping' here).